status/versions.go

```diff
@@ -308,22 +315,83 @@ func kubernetesVersion() (*ExternalServiceInfo, error) {
+// did that that change will matter, and the kiali pod could be bounced as a workaround.
+var isMixerDisabled *bool
+
+// IsMixerDisabled returns true if Telemetry V2 is enabled (mixer is not collecting metrics)
+// TODO: This test can be removed when Kiali stops supporting Istio versions with Mixer Telemetry
+func IsMixerDisabled() bool {
+	if isMixerDisabled != nil {
+		return *isMixerDisabled
+	}
+
+	k8sConfig, err := kubernetes.ConfigClient()
+	if err != nil {
+		log.Warningf("IsMixerDisabled: Cannot create config structure Kubernetes Client.")
+		return false
+	}
 
-// IstioSupportsCanonical returns true if Istio version can be determined and is >= 1.5.
-// TODO: This test can be removed when Kiali stops supporting any Istio versions < 1.5
-func IstioSupportsCanonical() bool {
-	if istioSupportsCanonical != nil {
-		return *istioSupportsCanonical
+	k8s, err := kubernetes.NewClientFromConfig(k8sConfig)
+	if err != nil {
+		log.Warningf("IsMixerDisabled: Cannot create Kubernetes Client.")
+		return false
 	}
 
-	istioVersion, err := istioVersion()
+	cfg := config.Get()
+	istioConfig, err := k8s.GetConfigMap(cfg.IstioNamespace, ISTIO_CONFIGMAP_NAME)
 	if err != nil {
+		log.Warningf("IsMixerDisabled: Cannot retrieve Istio ConfigMap.")
 		return false
 	}
-	valid := validateVersion(">= 1.5", istioVersion.Version)
-	istioSupportsCanonical = &valid
-	return *istioSupportsCanonical
+
+	meshConfigYaml, ok := istioConfig.Data["mesh"]
+	log.Tracef("meshConfig: %v", meshConfigYaml)
+	if !ok {
+		log.Warningf("IsMixerDisabled: Cannot find Istio mesh configuration.")
+		return false
+	}
+
+	meshConfig := istioMeshConfig{}
+	err = yaml.Unmarshal([]byte(meshConfigYaml), &meshConfig)
+	if err != nil {
+		log.Warningf("IsMixerDisabled: Cannot read Istio mesh configuration.")
+		return false
+	}
+
+	log.Infof("IsMixerDisabled: %t", meshConfig.DisableMixerHttpReports)
+
+	// References:
+	//   * https://github.com/istio/api/pull/1112
+	//   * https://github.com/istio/istio/pull/17695
+	//   * https://github.com/istio/istio/issues/15935
+	isMixerDisabled = &meshConfig.DisableMixerHttpReports
+	return *isMixerDisabled
+}
+
+// set this one time, it is very unlikely that the istio version will change without a kiali pod restart, or if it
+// did that that version change will matter, and the kiali pod could be bounced as a workaround.
+var istioSupportsCanonical *bool
+
+// AreCanonicalMetricsAvailable returns true if canonical labels are present in Istio Telemetry.
+// TODO: This test can be removed when Kiali stops supporting Istio versions with Mixer Telemetry
+func AreCanonicalMetricsAvailable() bool { // AreCanonicalMetricsAvailable() bool {
+	if istioSupportsCanonical == nil {
+		// First, check Istio version because canonical labels were first introduced in Istio v1.5. Prior
+		// Istio versions won't have canonical labels in metrics regardless of active Telemetry version.
+		istioVersion, err := istioVersion()
+		if err != nil {
+			return false
```

Comment:
it may seem countr-intuitive given my other comments but I think this is still the correct default.  If we don't know for sure that canonical is supported then we should fall back to app/version because they will exist in older installs and will still be there (deprecated) for several more versions.